x-pack/plugins/security_solution/public/common/components/event_details/get_alert_summary_rows.tsx

```diff
@@ -322,3 +306,63 @@ export const getSummaryRows = ({
       }, [])
     : [];
 };
+
+/**
+ * Enriches the summary data for threshold terms.
+ * For any given threshold term, it generates a row with the term's name and the associated value.
+ */
+function enrichThresholdTerms(
+  { values: termsFieldArr }: TimelineEventsDetailsItem,
+  data: TimelineEventsDetailsItem[],
+  description: EnrichedFieldInfo
+) {
+  const termsValueItem = data.find((d) => d.field === THRESHOLD_TERMS_VALUE);
+  const termsValueArray = termsValueItem && termsValueItem.values;
+
+  // Make sure both `fields` and `values` are an array and that they have the same length
+  if (
+    Array.isArray(termsFieldArr) &&
+    termsFieldArr.length > 0 &&
+    Array.isArray(termsValueArray) &&
+    termsFieldArr.length === termsValueArray.length
```

Comment:
consider adding additional unit tests to cover these cases

Reply:
✔️  https://github.com/elastic/kibana/pull/125376/commits/d343a540e2824191c9676d95b71cd6c41f21b051 https://github.com/elastic/kibana/pull/125376/commits/7f4be4d3047c23ee9abab73545eea8859c0fa82e